emcc.py

```diff
@@ -1472,6 +1475,11 @@ def check(input_file):
           # table, as is standard in wasm, and not asm.js split ones.
           shared.Settings.EMULATED_FUNCTION_POINTERS = 1
 
+    if shared.Settings.WASM2JS:
+      if not shared.Settings.WASM_BACKEND:
+        exit_with_error('wasm2js is only available in the upstream wasm backend path')
```

Comment:
Showerthought: we do these `if condition: exit_with_error` often enough that maybe we should want an `exit_if(condition, message)` function

@sbc100 yay nay?

Reply:
meh.  i think this is fine.